packages/ic-asset-certification/src/lib.rs

```diff
@@ -1,530 +1,770 @@
-useful when direct access to the
-[HttpCertificationTree](ic_http_certification::HttpCertificationTree) is required
-for certifying [HttpRequest](ic_http_certification::HttpRequest)s and
-[HttpResponse](ic_http_certification::HttpResponse)s outside of the [AssetRouter].
-
-```rust
-use std::{cell::RefCell, rc::Rc};
-use ic_http_certification::HttpCertificationTree;
-use ic_asset_certification::AssetRouter;
-
-let mut http_certification_tree: Rc<RefCell<HttpCertificationTree>> = Default::default();
-let mut asset_router = AssetRouter::with_tree(http_certification_tree.clone());
-```
-
-## Serving assets
-
-Assets can be served by calling the `serve_asset` method on the `AssetRouter`.
-This method will return a response, a witness and an expression path.
-
-```rust
-use ic_http_certification::HttpRequest;
-use ic_asset_certification::{Asset, AssetConfig, AssetFallbackConfig, AssetRouter};
-
-let mut asset_router = AssetRouter::default();
-
-let asset = Asset::new(
-    "index.html",
-    b"<html><body><h1>Hello World!</h1></body></html>".as_slice(),
-);
-
-let asset_config = AssetConfig::File {
-    path: "index.html".to_string(),
-    content_type: Some("text/html".to_string()),
-    headers: vec![
-        ("Cache-Control".to_string(), "public, no-cache, no-store".to_string()),
-    ],
-    fallback_for: vec![AssetFallbackConfig {
-        scope: "/".to_string(),
-    }],
-    aliased_by: vec!["/".to_string()],
-    encodings: vec![],
-};
-
-let http_request = HttpRequest::get("/").build();
-
-asset_router.certify_assets(vec![asset], vec![asset_config]).unwrap();
-
-let (mut response, witness, expr_path) = asset_router.serve_asset(&http_request).unwrap();
-```
-
-Some additional steps are then required to prepare the response for sending:
-
-- get the canister's certificate data
-- CBOR-encode the witness
-- CBOR-encode the expression path
-- Add the certificate header to the response
-
-```ignore
-const IC_CERTIFICATE_HEADER: &str = "IC-Certificate";
-fn add_certificate_header(response: &mut HttpResponse, witness: &HashTree, expr_path: &[String]) {
-    let certified_data = data_certificate().expect("No data certificate available");
-    let witness = cbor_encode(witness);
-    let expr_path = cbor_encode(&expr_path);
-
-    response.headers.push((
-        IC_CERTIFICATE_HEADER.to_string(),
-        format!(
-            "certificate=:{}:, tree=:{}:, expr_path=:{}:, version=2",
-            BASE64.encode(certified_data),
-            BASE64.encode(witness),
-            BASE64.encode(expr_path)
-        ),
-    ));
-}
-
-// Encoding
-fn cbor_encode(value: &impl Serialize) -> Vec<u8> {
-    let mut serializer = serde_cbor::Serializer::new(Vec::new());
-    serializer
-        .self_describe()
-        .expect("Failed to self describe CBOR");
-    value
-        .serialize(&mut serializer)
-        .expect("Failed to serialize value");
-    serializer.into_inner()
-}
-
-add_certificate_header(&mut response, &witness, &expr_path);
-```
-*/
+//!
```

Comment:
This file had issues with strings like `"**/*.js"`, removing the multiline doc string in favor of multiple singleline doc strings fixed the issue. I opened an issue against Rust here: https://github.com/rust-lang/rust/issues/129685

For review purposes, all changes here match exactly the changes in `README.md`

Reply:
nit: is it worthwhile to have two copies of the documentation in a package?  Wouldn't it be sufficient to have just a comprehensive `README.md` and succint per-function comments? (Maintaining both copies in sync seems quite challenging, no?)

Reply:
As we discussed offline, let's keep it for this PR and review alternatives over the next days